modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ParameterGroupArgumentResolver.java

```diff
@@ -8,20 +8,25 @@
 
 import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
 import static org.mule.runtime.api.util.Preconditions.checkState;
+
 import org.mule.runtime.api.exception.MuleRuntimeException;
 import org.mule.runtime.api.util.LazyValue;
 import org.mule.runtime.extension.api.runtime.operation.ExecutionContext;
 import org.mule.runtime.module.extension.api.runtime.privileged.EventedExecutionContext;
 import org.mule.runtime.module.extension.internal.loader.ParameterGroupDescriptor;
 import org.mule.runtime.module.extension.internal.runtime.objectbuilder.ParameterGroupObjectBuilder;
+import org.mule.runtime.module.extension.internal.util.ReflectionCache;
 
 public final class ParameterGroupArgumentResolver<T> implements ArgumentResolver<T> {
 
   private final ParameterGroupDescriptor group;
 
-  public ParameterGroupArgumentResolver(ParameterGroupDescriptor group) {
+  private final ReflectionCache reflectionCache;
+
+  public ParameterGroupArgumentResolver(ParameterGroupDescriptor group, ReflectionCache reflectionCache) {
     checkState(group.getType().isInstantiable(), "Class %s cannot be instantiated.");
     this.group = group;
+    this.reflectionCache = reflectionCache;
```

Comment:
value resolver should support the @Inject mechanism to receive the cache, maybe that simplifies the logic instead of having the parameter